### PR TITLE
Feature Bubble Sort

### PR DESCRIPTION
**Pull request by carloso0114**
In this project we manage to create two methods that sort an array of numbers and strings.
The tools that we use were:
-Ruby
-Rubocop

It consists of two different methods, the second one includes a yield and a block statement with an argument as required. 
Only works with an array of strings
The first method works only with an array of integers.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,26 @@
...
+## Built With
+- Ruby
+- Rubocop
+- Git
+
+## How to run it
+To get a local copy up and running follow these simple steps.
+* If you're using latest version of git.
+* Clone this url https://github.com/Milypm/Ruby-Bubble-Sort/tree/main on your local host.
+* Open the ruby file with the terminal running the command line "ruby bubble-sort.rb"
```
Comment (kitpao) [inline README.md]: - [x] I copy pasted your instruction and it gave me an error, please fix it. Remember that `-` is not the same as `_`

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if arr[i] > arr[i + 1]
+    i += 1
+    if i == length - 1
+      i = 0
+      length -= 1
+    end
+  end
+  arr
+end
+puts bubble_sort([4, 3, 5, 1, 2])
```
Comment (kitpao) [inline bubble_sort.rb]: - [x] About good practices, please do not commit debugging code (generally include `puts`, `print` or `p`). Later in the course, you will learn how to add automated tests where you will be able to put your examples ;) Please delete this line
image

File: bubble_sort.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    arr[i], arr[i + 1] = arr[i + 1], arr[i] if yield(arr[i], arr[i + 1]).positive?
+    i += 1
+  end
+  arr
+end
+
+a = bubble_sort_by(%w[hi hello hey]) do |left, right|
+  left.length - right.length
+end
+puts a
```
Comment (kitpao) [inline bubble_sort.rb]: - [x] About good practices, please do not commit debugging code (generally include `puts`, `print` or `p`). Later in the course, you will learn how to add automated tests where you will be able to put your examples ;) Please delete these lines
image

**Review by kitpao — CHANGES_REQUESTED** [review]
# 👋 Hello @Milypm and @carloso0114 

🌟 Great job with your logic, it works perfectly 💯 Your readme and pull request messages are very useful as well (you just have 1 detail in the Readme). 

📝 Please check the inline commits to see and fix the remaining issues ⏬ 

- Your commits are descriptive, but for next time try to be consistent and write your commits in the imperative tense, as stated in [the article pointed in pathwright](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)

### ♻️ Status: Your project requires some additional changes ♻️
➡️ Please request another review once you are ready 📝 
___
##### 👉 If you have any doubts you can contact me on slack as @kitpao
##### 🌟  If you have feedback about my reviews I would appreciate it if you could leave a recommendation in LinkedIn 💌 or let me know in Slack if you see improvements to be done 😄  

🔗 __[Github](https://github.com/kitpao) | [Twitter](https://twitter.com/Kitpao1) | [LinkedIn](https://www.linkedin.com/in/kitzia-paola-vidal/) | [Medium](https://medium.com/@kitpao91)__

**Review by Georjane — APPROVED** [review]
Status: PR Approved ✔️✔️
Hi Team🙌
Congratulations on your Bubble Sort Project. 🎊🎉
Your project looks great and you implemented all changes required by the previous TSE. You can go ahead and merge this PR. 👍👍

Goodluck and Happy Coding 💻

**Pull request merged**